task1/server.js

```diff
@@ -13,61 +17,57 @@ http
       time: new Date()
     };
 
-    if (fs.existsSync(path)) {
-      writeToJson(path, requestInfo);
-    } else {
-      createNewFile(path, requestInfo);
+    try {
+      await writeToJson(path, requestInfo);
+    }
+    catch(err) {
+      console.error(err);
+      return;
     }
 
     res.writeHead(200, { "Content-Type": "application/json" });
 
     if (req.url === "/get-all-logs" && req.method === "GET") {
-      return res.end(JSON.stringify(readFromFile(path)));
+      return res.end(JSON.stringify(await readFromFile(path)));
     }
 
-    let urlHasQuery = (Object.keys(queryObject).length === 0) ? false : true;
+    let urlHasQuery = !(Object.keys(queryObject).length === 0);
     if (urlHasQuery) {
       const from = queryObject.from;
       const to = queryObject.to;
-      let array = getDataRanged(path, from, to);
+      let array = await getDataRanged(path, from, to);
       return res.end(JSON.stringify(array));
```

Comment:
im not sure, but shouldnt  we send logs only if its /get-all-logs route?

Reply:
No, we should have a route for getting logs for specific date range.